### PR TITLE
Implement TOC window auto-fit

### PR DESCRIPTION
**Pull request by alexconst**
Allow the user, to set an option, to have the TOC window shrink its size
in order to auto-fit to contents.

### CODE REVIEW

**Comment by alexconst** [comment]
It's failing on the test I implemented which checks vim window size.
So Travis small window width is breaking things here.
What's the work around for this?

**Comment by shirosaki** [comment]
Thanks for the patch with tests.
I worked around to limit max width by getting current window width `winwidth(0)`.
https://github.com/plasticboy/vim-markdown/blob/master/test/python-folding.vader#L32

I found another way to change width on travis.
See https://github.com/travis-ci/travis-ci/issues/5407

File: test/toc-autofit.vader [hunk 1]
```diff
@@ -0,0 +1,57 @@
+" Tests toc window auto-fit to longest header, but without exceeding half screen.
+
+Before:
+  source ../after/ftplugin/markdown.vim
+
+After:
+  setlocal foldexpr=0
+  setlocal foldmethod=manual
+
```
Comment (shirosaki) [inline test/toc-autofit.vader]: Above Before After part is needed only for folding. In this case not needed.

File: test/toc-autofit.vader [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  set number
+  let g:vim_markdown_toc_autofit = 1
+  let line = '###### chap 12.1.1.1.1.1'
+  AssertEqual getline('33'), line
+  :Toc
+  let expected_width = len(line) + 2*5 + 1 + 3 - 7
+  AssertEqual &winwidth, expected_width
+" 2 spaces * 5 additional header levels + 1 space for first header +
+" 3 spaces for line numbers - 7 chars ('###### ') that don't show up on the TOC
+
```
Comment (shirosaki) [inline test/toc-autofit.vader]: Adding `:lclose` after `:Toc` would be better not to break other tests.

**Comment by shirosaki** [comment]
Merged. Looks nice to me.
It may be good that default behavior is auto-fit.

**Pull request merged**